zeppelin-web-angular/src/app/pages/workspace/notebook/notebook.component.ts

```diff
@@ -109,6 +109,10 @@ export class NotebookComponent extends MessageListenersManager implements OnInit
 
   @MessageListener(OP.INTERPRETER_BINDINGS)
   loadInterpreterBindings(data: MessageReceiveDataTypeMap[OP.INTERPRETER_BINDINGS]) {
+    this.listOfNotebookParagraphComponent.forEach(item => {
+      item.notebookParagraphCodeEditorComponent.editorSettingTriggerAllowed = true;
```

Comment:
We should handle some case where `notebookParagraphCodeEditorComponent` is undefined.
Also, should we trigger editor settings for all paragraphs here? Would calling `this.setParagraphMode(true)` in `editor.onDidChangeModelContent()` be sufficient to get editor settings only to the paragraph whose interpreter has changed?

Reply:
In testing, the interpreter setting works correctly when `this.setParagraphMode(true)` is executed inside `editor.onDidChangeModelContent()`.

Since this runs **before** the code that executes when receiving `INTERPRETER_BINDINGS` above (`editor.onDidChangeModelContent()`), the code that runs after `editorSettingTriggerAllowed = true` was modified to directly call `getEditorSetting()` instead of using `setParagraphMode(true)` when receive `PARAGRAPH` or `INTERPRETER_BINDINGS`. Because if the interpreter is already set inside `this.setParagraphMode(true)`, `getEditorSetting()` will not be executed due to the condition, and since `setParagraphMode(true)` has already run, there is no need to execute it again.